### DISCUSSION TITLE
Allow creation of records for unknown collections.

### DISCUSSION THREAD

**Discussion by mikuhl-dev** [bluesky-social/atproto#1149] [General Discussion]
If clients are to implement their own features, allowing the creation of records on unknown collections is needed. Unknown collections will eventually end up on on the server anyway if someone moves their data from one with a feature to one without. Ideally the server should not delete the data it doesn't understand.

Request:
```
curl --request POST \
  --url https://bsky.social/xrpc/com.atproto.repo.createRecord \
  --header 'Content-Type: application/json' \
  --data '{
	"repo": "mikuhl.bsky.social",
	"collection": "org.bitcoin.lightning.profile",
	"rkey": "self",
	"record": {
		"lud16": "mikuhl@ln.tips"
	}
}'
```
Response:
```json
{
	"error": "InvalidRequest",
	"message": "Lexicon not found: lex:org.bitcoin.lightning.profile"
}
```
Request:
```
curl --request POST \
  --url https://bsky.social/xrpc/com.atproto.repo.putRecord \
  --header 'Content-Type: application/json' \
  --data '{
	"repo": "mikuhl.bsky.social",
	"collection": "org.bitcoin.lightning.profile",
	"rkey": "self",
	"record": {
		"lud16": "mikuhl@ln.tips"
	}
}'
```
Response:
```json
{
	"error": "InvalidRequest",
	"message": "Temporarily only accepting puts for app.bsky.actor.profile/self."
}
```

**Comment by bnewbold** [comment]
This is still an open issue, and is core to how atproto works.

We will definitely update the PDS reference implementation, and reconfigure our production PDS instance, to allow other Lexicons in the not too distant future. We just don't have the bandwidth to open this up in a well-supported way ourselves right now. Other folks have already patched the PDS or implemented their own to get around this restriction and that is fine.
